tests/unit/acquisition/test_function.py

```diff
@@ -495,5 +489,69 @@ def _build_with_sampler(
             )
             return raise_
 
-    data = Dataset(tf.zeros([0, 1]), tf.zeros([0, 2]))
-    _Acq(100).prepare_acquisition_function(data, _ArbitraryDimTwoOutputModel())
+    data = Dataset(tf.zeros([0, 2], dtype=tf.float64), tf.zeros([0, 2], dtype=tf.float64))
+    _Acq(1000).prepare_acquisition_function(data, _dim_two_gp())
+
+
+@pytest.mark.parametrize("sample_size", [0, -2])
+def test_batch_reparametrization_sampler_raises_for_invalid_sample_size(sample_size: int) -> None:
+    with pytest.raises((ValueError, tf.errors.InvalidArgumentError)):
+        BatchReparametrizationSampler(sample_size, _dim_two_gp())
+
+
+@random_seed
+def test_batch_reparametrization_sampler_samples_approximate_mean_and_covariance() -> None:
+    model = _dim_two_gp()
+    sample_size = 10_000
+    leading_dims = [3]
+    batch_size = 4
+    xs = tf.random.uniform(leading_dims + [batch_size, 2], maxval=1.0, dtype=tf.float64)
+    samples = BatchReparametrizationSampler(sample_size, model).sample(xs)
+
+    assert samples.shape == leading_dims + [sample_size, batch_size, 2]
+
+    samples_mean = tf.reduce_mean(samples, axis=-3)
+    samples_covariance = tf.transpose(
+        tfp.stats.covariance(samples, sample_axis=-3, event_axis=-2), [0, 3, 1, 2]
+    )
+
+    model_mean, model_cov = model.predict_joint(xs)
+
+    npt.assert_allclose(samples_mean, model_mean, rtol=0.02)
+    npt.assert_allclose(samples_covariance, model_cov, rtol=0.04)
```

Comment:
it took some real searching to find a batch size that would give this level of accuracy, and it's not even that good. I'm not sure what I need to do to improve it. I probably need to look again at the kernel

Reply:
ah, rbf's not suitable for quadratic curves

Reply:
With that many samples, the mean should be very accurate. Have you looked at what happens if you use 1e7 samples? Not for the actual test that would take too long but to make sure there's not error.

One possible explanation is that the covariance matrix contains values very close to zero (which would trigger a warning although the approximation is good). This would happen if the spread of xs is much larger than the lengthscale. Or conversely, the covariance matrix is ill-conditioned because the lengthscale is very large. Then the sampling introduces some numerical errors.